### PR TITLE
Windows: Deprecate moonjit support

### PR DESCRIPTION
**Pull request by wrowe**
Commit Message: Windows: Deprecate moonjit support
Additional Description:

See https://github.com/envoyproxy/envoy/issues/13539 

This is not the entire resolution of the issue, only a simplification

Risk Level: low
Testing: minimal
Docs Changes: n/a here (see c711935)
Release Notes: n/a here (see c711935)

### CODE REVIEW

**Comment by wrowe** [comment]
@htuch we can glue this all back together however you like, I wasn't going to deal with merge breakage to master this morning. Just watching the Windows CI result specifically. Whether you want to deprecate moonjit in this PR is your call, or I will cherry pick that into a new PR.

**Comment by htuch** [comment]
@wrowe thanks, I've patched this into https://github.com/envoyproxy/envoy/pull/13474. Fine to take the MoonJit removal for Windows.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -283,13 +283,15 @@ DEPENDENCY_REPOSITORIES_SPEC = dict(
...
-        version = "2.1.0-beta3",
-        sha256 = "409f7fe570d3c16558e594421c47bdd130238323c9d6fd6c83dedd2aaeb082a8",
+        # The last release version, 2.1.0-beta3 has a number of CVEs filed
+        # against it. These may not impact correct non-malicious Lua code, but for prudence we bump.
+        version = "e9af1abec542e6f9851ff2368e7f196b6382a44c",
+        sha256 = "186cd55c0e321aa2ec3c0f9220e31cc17479ec6222db515ec8b902f9275a9616",
         strip_prefix = "LuaJIT-{version}",
-        urls = ["https://github.com/LuaJIT/LuaJIT/archive/v{version}.tar.gz"],
+        urls = ["https://github.com/LuaJIT/LuaJIT/archive/{version}.tar.gz"],
+        last_updated = "2020-10-09",
```
Comment (moderation) [inline bazel/repository_locations.bzl:292]: This commit is 2020-09-29 and `last_updated` should reflect that. Consider bumping to latest commit `1d8b747c161db457e032a023ebbff511f5de5ec2` `2020-10-12`. Duplicate comment from https://github.com/envoyproxy/envoy/pull/13474#pullrequestreview-507672658

**Comment by wrowe** [comment]
Holding on merge of https://github.com/envoyproxy/envoy/pull/13474

/wait

**Comment by wrowe** [comment]
cc @envoyproxy/windows-dev just FYI

No reason to hold off. The windows port makes no reference to toggling moonjit. This patch simply drops all workarounds per the cited issue 13539 to make the residual patch much simpler, between now and decommissioning moonjit support altogether.

Converting this to a ready state, since it has no overlap with the current bump to luajit https://github.com/envoyproxy/envoy/pull/13474

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**